### PR TITLE
added all code of nn

### PR DESCRIPTION
**Pull request by Daniel231**

### CODE REVIEW

File: src/util/matchToKartoffel.js [hunk 1]
```diff
@@ -221,6 +221,72 @@ const match_ads = (obj) => {
...
+                break;
+            //hierarchy and job
+            case fn.adNN.hierarchy:
+                let hr = obj[rawKey].includes("\\") ? obj[rawKey].substring(0, obj[rawKey].lastIndexOf('\\')).trim().split('\\') : obj[rawKey].substring(0, obj[rawKey].lastIndexOf('/')).trim().split('/');
+                if (hr[0] == "") {
+                    delete obj[rawKey];
+                    break;
+                }
+                hr[0] === fn.rootHierarchy ? null : hr.unshift(fn.rootHierarchy);
+                hr.splice((hr.length - 1), 1);
```
Comment (weizmm) [inline src/util/matchToKartoffel.js]: remove this line baecause of using lastIndex command

File: src/util/matchToKartoffel.js [hunk 2]
```diff
@@ -221,6 +221,72 @@ const match_ads = (obj) => {
...
+                        job = job.replace(obj[fn.adNN.fullName],"").trim()
+                    }
+                    obj.job = job
+                }
+
+                (rawKey === "hierarchy") ? null : delete obj[rawKey];
+                break;
+            //personalNumber
+            case fn.adNN.sAMAccountName:
+                let uniqueNum = obj[rawKey].toLowerCase().includes(fn.adNN.extension) ? obj[rawKey].toLowerCase().replace(fn.adNN.extension, "") : break;
```
Comment (weizmm) [inline src/util/matchToKartoffel.js]: print warning log if there is not uniqeNum field

File: src/util/matchToKartoffel.js [hunk 3]
```diff
@@ -221,6 +221,72 @@ const match_ads = (obj) => {
...
+                }
+                hr[0] === fn.rootHierarchy ? null : hr.unshift(fn.rootHierarchy);
+                hr.splice((hr.length - 1), 1);
+                obj.hierarchy = hr.join("/");
+                obj.hierarchy = obj.hierarchy.replace(new RegExp('\u{200f}', 'g'), '');
+
+                // Getting job
+                if(obj[rawKey].includes("-")) {
+                    if(obj[rawKey].includes("\\")) {
+                        job = obj[rawKey].substring(obj[rawKey].lastIndexOf("\\") + 1).replace(/-/g,"")
```
Comment (weizmm) [inline src/util/matchToKartoffel.js]: add trim

File: src/util/matchToKartoffel.js [hunk 4]
```diff
@@ -221,6 +221,72 @@ const match_ads = (obj) => {
...
+                    } else {
+                        job = obj[rawKey].substring(obj[rawKey].lastIndexOf("/") + 1).replace(/-/g,"")
```
Comment (weizmm) [inline src/util/matchToKartoffel.js]:
```suggestion
                        job = obj[rawKey].substring(obj[rawKey].lastIndexOf("/") + 1).replace(/-/g,"").trim()
```

File: src/util/matchToKartoffel.js (hunk 1 above)
Comment (Daniel231) [reply src/util/matchToKartoffel.js]: removed

**Pull request merged**